File: src/gestorAplicacion/servicios/Formula.java
/* Clase que crea las formulas que se han generado para un paciente.
Autores: Diego Andres Gracia Granados, Daniel Giraldo Vanegas, Elian David Velandia Riveros, Juan Camilo Gutierrez Martinez
y Santiago Arboleda Acevedo */
package gestorAplicacion.servicios;

import gestorAplicacion.administracion.Medicamento;
import gestorAplicacion.personas.Doctor;
import gestorAplicacion.personas.Paciente;

import java.text.Normalizer;
import java.util.ArrayList;

public class Formula extends Servicio {
    // Atributos
    private ArrayList<Medicamento> listaMedicamentos = new ArrayList<Medicamento>();
    private Doctor doctor;

    // Constructor
    public Formula(ArrayList<Medicamento> listaMedicamentos, Doctor doctor, Paciente paciente) {
        super(paciente);
        this.listaMedicamentos = listaMedicamentos;
        this.doctor = doctor;
    }
    // Sobrecarga de constructor
    public Formula(Paciente paciente){
        this(null, null, paciente);
    }

    // Getters y setters
    public ArrayList<Medicamento> getListaMedicamentos() {
        return listaMedicamentos;
    }

    public void setListaMedicamentos(ArrayList<Medicamento> listaMedicamentos) {
        this.listaMedicamentos = listaMedicamentos;
    }

    public Doctor getDoctor() {
        return doctor;
    }

    public void setDoctor(Doctor doctor) {
        this.doctor = doctor;
    }
    // Metodos

    // Método sobreescrito
    @Override
    public void validarPago(Paciente paciente, long idServicio) {
        for (Formula formula :
                paciente.getHistoriaClinica().getListaFormulas()) {
            if (formula.getIdServicio() == idServicio)
                formula.setEstadoPago(true);
        }
    }

    // Metodo que imprime una descripcion del servicio
    @Override
    public String descripcionServicio() {
        return idServicio + " - Formula prescrita por " + doctor.getNombre();
    }

    // toString
    @Override
    public String toString() {
        return "Hola "+paciente.getNombre()+"\nEstos son tus medicamentos formulados:\n"+listaMedicamentos;
    }
}

File: Python/src/gestor_aplicacion/servicios/formula.py
# Autores: Diego Andres Gracia Granados, Daniel Giraldo Vanegas, Elian David Velandia Riveros, Juan Camilo Gutierrez Martinez
# y Santiago Arboleda Acevedo
from src.gestor_aplicacion.servicios.servicio import Servicio

# Clase que crea las formulas que se han generado para un paciente.
class Formula(Servicio):

    # Atributos y constructor
    def __init__(self, paciente):
        super().__init__(paciente)
        self._lista_meds = []
        self._doctor = None

    # Metodos

    # Metodo que imprime una descripcion del servicio
    def descripcion_servicio(self):
        return f"{self._id_servicio} - Formula prescrita por {self._doctor}"

    # Métodos que busca el estado de pago de una cita médica y lo cambia
    # Recibe un objeto de la clase Paciente y un entero
    def confirmar_pago(self, paciente, id_servicio):
        for formula in paciente.historia_clinica.lista_formulas:
            if formula.id_servicio == id_servicio:
                formula.estado_pago = True
                break

    @property
    def doctor(self):
        return self._doctor

    @doctor.setter
    def doctor(self, value):
        self._doctor = value

    @property
    def lista_meds(self):
        return self._lista_meds

    @lista_meds.setter
    def lista_meds(self, value):
        self._lista_meds = value
